chemicaltoolbox/rdkit/splitter.py

```diff
@@ -0,0 +1,51 @@
+import argparse
```

Comment:
I guess this tool is not needed as it can be done by other tools?

Reply:
I need to look at the alternatives

Reply:
I don't see any exact alternatives at present. 
The `Split file` tool has a few built in formats that it understands, but SDF is not one of them. If you use the `Text files` option for the format it doesn't do the splitting correctly (in fact it doesn't split at all you get n identical files.

The `Prepare ligands for docking` tool could be extended to handle this. Currently it seem to output each record as a separate file, but it might get a bit confusing as some of the formats it supports (e.g. molfile) do not support multiple records in a single file. 

My `Splitter` tool is based on a much more comprehensive tool that can do much more than splitting (e.g filtering, transforming data etc.) which could also be added at some stage and would be useful. See https://github.com/InformaticsMatters/pipelines-utils-rdkit/blob/master/src/python/pipelines_utils_rdkit/filter.py

So unless I'm not seeing some obvious tool that can handle this then I prefer to keep the splitter tool and then to extend it at a later stage.

Reply:
Hi Tim, for me the tool is working, but you need to use "Generic", there is even a test included that makes sure it works for SD files. Here is a history for you to double check: https://usegalaxy.eu/u/bgruening/h/file-splitting-for-tim.

> The Prepare ligands for docking tool could be extended to handle this. Currently it seem to output each record as a separate file, but it might get a bit confusing as some of the formats it supports (e.g. molfile) do not support multiple records in a single file.

Yes we could do this, but in general we try to follow Unix philosophy as much as we can. Simple tools that do one thing well - exceptions are usability.

> My Splitter tool is based on a much more comprehensive tool that can do much more than splitting (e.g filtering, transforming data etc.) which could also be added at some stage and would be useful. See https://github.com/InformaticsMatters/pipelines-utils-rdkit/blob/master/src/python/pipelines_utils_rdkit/filter.py

Same here. I think this functionality is available in other tools already. We have a ton of filtering and transformation tools. In the end its also a problem if Findability. People searching for filtering are not looking at a splitting tool and vice versa. So I would keep functionality as much as possible separated. But yes, I know it has advantages as seen with Vina to merge tools - I guess we need always to discuss this on a case-to-case basis.

Reply:
@bgruening The splitting does not seem to be correct. It does not split at the end of a record. Look as the start of the 3rd part - it starts in the middle of the data section.
Also, splitting is done on the basis of how many files your want, not the number of records you want in each file.